src/main/java/Duke.java

```diff
@@ -1,10 +1,56 @@
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    String getResponse(String input) {
+        return "Duke heard: " + input;
+    }
+
+    public Duke() {
+
+    }
+
+    private static final String saveLoadFilePath = "listSaveData.txt";
+
+    private Duke(String filepath) {
+        // initialise variables
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage(filepath);
+        try {
+            tasks = storage.load();
+        } catch (Exception e) {
+            ui.print(e.getMessage());
+            tasks = new TaskList();
+        }
+    }
+
+    private void run() {
+        try (Scanner scanner = new Scanner(System.in);
+             PrintStream out = new PrintStream(System.out, true, StandardCharsets.UTF_8)
```

Comment:
Oo... may i ask what's print stream used for?

Reply:
when I used to run it on the command line, the cross and tick characters don't show. A trick I found online is to directly set the PrintStream